addons/calendar/static/tests/calendar_notification_tests.js

```diff
@@ -0,0 +1,205 @@
+
+    testConfig = getActionManagerTestConfig();
+    testConfig.serviceRegistry.add('calendarNotification', calendarNotificationService);
+
+    testConfig.mainComponentRegistry = new Registry();
+    testConfig.mainComponentRegistry.add("NotificationContainer", NotificationContainer);
+
+    testConfig.browser.setTimeout = (fn) => fn();
+    testConfig.browser.clearTimeout = () => {};
+  });
+
+  QUnit.test('can listen on bus and displays notifications in DOM and click OK', async (assert) => {
+    assert.expect(5);
+
+    const mockRPC = (route, args) => {
+      if (route === '/longpolling/poll') {
+        return Promise.resolve([{
+          id: 'prout',
+          channel: ['db_name', 'calendar.alarm', 7],
+          message: [
+            {
+                "alarm_id": 1,
+                "event_id": 2,
+                "title": "Meeting",
+                "message": "Very old meeting message",
+                "timer": 20 * 60,
+                "notify_at": '1978-04-14 12:45:00',
+            }
+          ],
+        }]);
+      }
+      if (route === '/calendar/notify') {
+        return Promise.resolve(true);
+      }
+      if (route === '/calendar/notify_ack') {
+        assert.step("notifyAck");
+        return Promise.resolve(true);
+      }
+    };
+
+    const webClient = await createWebClient({
+      testConfig,
+      legacyParams: { serviceRegistry: legacyServicesRegistry },
+      mockRPC,
+    });
+
+    await nextTick();
+
+    assert.containsOnce(webClient.el, '.o_notification_body');
+    assert.strictEqual(
+      webClient.el.querySelector('.o_notification_body .o_notification_content').textContent,
+      "Very old meeting message"
+    );
+
+    await click(webClient.el.querySelectorAll('.o_notification_buttons .btn')[0]);
+    assert.verifySteps(["notifyAck"]);
+    assert.containsNone(webClient.el, ".o_notification");
+
+    webClient.destroy();
+  });
+
+  QUnit.test('can listen on bus and displays notifications in DOM and click Detail', async (assert) => {
+    assert.expect(5);
+
+    const mockRPC = (route, args) => {
+      if (route === '/longpolling/poll') {
+        return Promise.resolve([{
+          id: 'prout',
+          channel: ['db_name', 'calendar.alarm', 7],
+          message: [
+            {
+                "alarm_id": 1,
+                "event_id": 2,
+                "title": "Meeting",
+                "message": "Very old meeting message",
+                "timer": 20 * 60,
+                "notify_at": '1978-04-14 12:45:00',
+            }
+          ],
+        }]);
+      }
+      if (route === '/calendar/notify') {
+        return Promise.resolve(true);
+      }
+    };
+
+    const fakeActionService = {
+      name: "action",
+      deploy() {
+        return {
+          doAction(actionId) {
+            assert.step(actionId);
+            return Promise.resolve(true);
+          },
+          loadState(state, options) {
```

Comment:
not sure you need this

Reply:
loadState is called when creating the webClient : 
https://github.com/odoo-dev/odoo/blob/1d56e53691304e8b0ef5d9c8ddfa3578e0cd3162/addons/web/static/src/webclient/webclient.js#L45